### ISSUE TITLE
WARNING - Clamscan reports database integrity tested BAD - SKIPPING

### ISSUE THREAD

**Issue by clamal** [extremeshok/clamav-unofficial-sigs#52]
Hi,
I installed the last version, I've this error when use cron in slackware;

WARNING - Clamscan reports xxxxxx database integrity tested BAD - SKIPPING

I tried to delete signature, but same result.
If I run manually the cron, sometimes this error disappear.

Best Regards.

**Comment by extremeshok** [comment]
which database file ?

**Comment by clamal** [comment]
The sanesecurity DB.

**Comment by clamal** [comment]
Mar 25 13:44:38 Sanesecurity Database & GPG Signature File Updates
Mar 25 13:44:38 Sanesecurity mirror site used: spamexperts-mirror.sanesecurity.com 185.66.251.102
Mar 25 13:44:39 Testing updated Sanesecurity database file: jurlbl.ndb
Mar 25 13:44:39 Clamscan reports Sanesecurity jurlbl.ndb database integrity tested BAD
Mar 25 13:44:39 Testing updated Sanesecurity database file: rogue.hdb
Mar 25 13:44:39 Clamscan reports Sanesecurity rogue.hdb database integrity tested BAD
Mar 25 13:44:39 Testing updated Sanesecurity database file: blurl.ndb
Mar 25 13:44:39 Clamscan reports Sanesecurity blurl.ndb database integrity tested BAD
Mar 25 13:44:39 Testing updated Sanesecurity database file: crdfam.clamav.hdb
Mar 25 13:44:39 Clamscan reports Sanesecurity crdfam.clamav.hdb database integrity tested BAD
Mar 25 13:44:39 Testing updated Sanesecurity database file: phishtank.ndb
Mar 25 13:44:39 Clamscan reports Sanesecurity phishtank.ndb database integrity tested BAD
Mar 25 13:44:39 Testing updated Sanesecurity database file: porcupine.hsb
Mar 25 13:44:39 Clamscan reports Sanesecurity porcupine.hsb database integrity tested BAD
Mar 25 13:44:39 No Sanesecurity database file updates found

**Comment by clamal** [comment]
If I run:
clamscan -d /usr/local/share/clamav-unofficial-sigs/dbs-ss/porcupine.hsb /usr/local/share/clamav-unofficial-sigs/configs/scan-test.txt

after a BAD result, I obtain:

/usr/local/share/clamav-unofficial-sigs/configs/scan-test.txt: OK

----------- SCAN SUMMARY -----------
Known viruses: 475
Engine version: 0.99.1
Scanned directories: 0
Scanned files: 1
Infected files: 0
Data scanned: 0.00 MB
Data read: 0.00 MB (ratio 0.00:1)
Time: 0.005 sec (0 m 0 s)

**Comment by extremeshok** [comment]
Its most likely selinux or the cron user's permissions.

**Comment by clamal** [comment]
Hi, selinux isn't installed. I try cron with user and root permissions, but same result.

**Comment by extremeshok** [comment]
Slackware ? Which version.

**Comment by clamal** [comment]
Slackware14.1

**Comment by clamal** [comment]
I will try to change "clamscan" with "/usr/local/bin/clamscan" in the script to test the cron.

**Comment by clamal** [comment]
With the full path for the clamscan executable in the "clamav-unofficial-sigs.sh", at this moment the error is disappeared.

**Comment by extremeshok** [comment]
Give me the output of the following, run as the cron user, without the quotes

"which clamscan"

Can I also have your os.conf

Thanks

**Comment by clamal** [comment]
The result is an empty file when cron runs alone.
If I run it manually, in the log file there is the path.
I'm using only the master.conf:
#clam_user="clamav"
#clam_group="clamav"
setmode="no"
clam_dbs="/usr/local/share/clamav"
clamd_pid="/var/run/clamd.pid"

**Comment by extremeshok** [comment]
replace your master.conf with the one from the script https://github.com/extremeshok/clamav-unofficial-sigs/blob/dev/config/master.conf
rename the os.slackware.conf to os.conf ( https://github.com/extremeshok/clamav-unofficial-sigs/blob/dev/config/os.slackware.conf )
edit/set your values in user.conf
It will work
btw. slackware is still stuck in the 90's

**Comment by extremeshok** [comment]
Verdict ?

**Comment by clamal** [comment]
Hi, I use the two files, but the error disappear if I use the full path in the script:

/usr/local/bin/clamscan

Best Regards.

**Comment by twouters** [comment]
Looks like `/usr/local/bin` isn't in your environments PATH.
Could you verify by running `env | grep PATH` as the user running `clamav-unofficial-sigs.sh`?
You could either change the PATH for this user by adding it to your bashrc or run `clamav-unofficial-sigs.sh` as follows:

~~~
PATH=$PATH:/usr/local/bin /bin/bash clamav-unofficial-sigs.sh
~~~

**Comment by extremeshok** [comment]
Are you running this on a server that has cpanel installed

**Comment by extremeshok** [comment]
Please test with the latest "DEV" version

**Comment by clamal** [comment]
On the Slack is installed webmin as panel.

**Comment by extremeshok** [comment]
Should be fixed in 5.0.5

Open if there are issues

**Comment by clamal** [comment]
Hi, with the DEV it's ok. I put che clamscan path in the master.conf.
Best Regards.
